transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttSslHandlerProvider.java

```diff
@@ -71,13 +82,14 @@ public SslHandler getSslHandler() {
             kmf.init(ks, keyStorePassword.toCharArray());
 
             KeyManager[] km = kmf.getKeyManagers();
-            TrustManager[] tm = tmFactory.getTrustManagers();
+            TrustManager x509wrapped = getX509TrustManager(tmFactory);
+            TrustManager[] tm = {x509wrapped};
             SSLContext sslContext = SSLContext.getInstance(TLS);
             sslContext.init(km, tm, null);
             SSLEngine sslEngine = sslContext.createSSLEngine();
             sslEngine.setUseClientMode(false);
             sslEngine.setNeedClientAuth(false);
-            sslEngine.setWantClientAuth(false);
+            sslEngine.setWantClientAuth(true);
```

Comment:
let's make this configurable. There are three options:
1) Using Access Token on unsecure connection (mqtt.ssl.key-store is not set)
2) Using Access Token over secure connection (mqtt.ssl.key-store is set, but mqtt.ssl.client-auth=false)
3) Using X.509 certificates over secure connection (mqtt.ssl.key-store is set and mqtt.ssl.client-auth=true)